app/controllers/application_controller.rb

```diff
@@ -0,0 +1,37 @@
+class ApplicationController < ActionController::API
+  before_action :authorized
+
+  def encode_token(payload)
+    JWT.encode(payload, 'yourSecret')
+  end
+
+  def auth_header
+    request.headers['Authorization']
+  end
+
+  def decoded_token
+    return nil unless auth_header
+
+    token = auth_header.split[1]
+    begin
+      JWT.decode(token, 'yourSecret', true, algorithm: 'HS256')
+    rescue JWT::DecodeError
+      nil
+    end
+  end
+
+  def logged_in_user
+    return nil unless decoded_token
+
+    user_id = decoded_token[0]['user_id']
+    @user = User.find_by(id: user_id)
+  end
+
+  def logged_in?
+    !!logged_in_user
```

Comment:
What did we need this method for? 

It appears to me you are trying to keep the user's authenticated state here in the backend. Unfortunately, that isn't the right way to do it as HTTP requests are supposed to be stateless.

The backend shouldn't be the one to monitor whether a user is logged in or not. The point of sending the front end a JWT back after every log in is so that JWT token can be sent back from subsequent front end requests. That way the back end will be able to accept requests for endpoints that are supposed to be for authenticated users.

Thus, this method "logged_in?" is unnecessary and in fact not very useful.